strongbox-web-core/src/main/java/org/carlspring/strongbox/utils/ArtifactControllerHelper.java

```diff
@@ -134,63 +194,160 @@ public static boolean isRangedRequest(HttpHeaders headers)
+                                                               contentType);
+                os.write(toByteArray(contentTypeHeader));
+
+                final String contentRangeHeader = String.format("%s: bytes %d-%d/%d",
+                                                                HttpHeaders.CONTENT_RANGE,
+                                                                start,
+                                                                end,
+                                                                inputLength);
+                os.write(toByteArray(contentRangeHeader));
+
+                os.write(toByteArray(""));
+
+                // Check if it is allowed to read the stream more than once.
+                if (bis.markSupported())
+                {
+                    int readLength;
+                    byte[] buffer = new byte[DEFAULT_BUFFER_SIZE];
+                    long toRead = length;
+
+                    long markLimit = Math.max(toRead, DEFAULT_BUFFER_SIZE) + 1L;
+                    int markLimitInt = Math.toIntExact(markLimit);
+                    // Needed for reading the stream more than once.
+                    bis.mark(markLimitInt);
+
+                    // Skip to the byte range offset.
+                    bis.skip(start);
+
+                    while ((readLength = bis.read()) != -1)
+                    {
+                        toRead -= readLength;
+
+                        // Range length is greater than the buffer length.
+                        if (toRead > 0)
+                        {
+                            os.write(buffer, 0, readLength);
+                            os.flush();
+
+                            totalBytes += readLength;
+                        }
+                        else
+                        {
+                            os.write(buffer, 0, (int) toRead + readLength);
+                            os.flush();
+
+                            totalBytes += (toRead + readLength);
+                            break;
+                        }
+                    }
+
+                    // Needed for reading the stream more than once.
+                    bis.reset();
+                }
             }
-            String checksumName = String.format("Checksum-%s",
-                                                e.getKey().toUpperCase().replaceAll("-", ""));
-            response.setHeader(checksumName,
-                               checksumValue);
-        });
-        
+
+            os.write(toByteArray(""));
+            os.write(toByteArray("--" + MULTIPART_BOUNDARY + "--"));
+            os.flush();
+
+            response.setHeader(HttpHeaders.CONTENT_LENGTH, Long.toString(totalBytes));
+            response.flushBuffer();
+        }
+    }
+
+    private static byte[] toByteArray(String string)
+    {
+        return (string.concat(System.lineSeparator())).getBytes(Charset.defaultCharset());
```

Comment:
`System.lineSeparator()`: standard HTTP end-of-line marker is `CRLF` (https://www.w3.org/Protocols/rfc2616/rfc2616-sec2.html#sec2.2) so we can't be system dependent here

Reply:
`Charset.defaultCharset()` - again, system specific. Please use explicitly `Charset.forName("UTF-8")`